modules/programs/firefox/mkFirefoxModule.nix

```diff
@@ -202,6 +203,71 @@ let
       (pkgs.wrapFirefox.override { config = bcfg; }) package { };
 
   bookmarkTypes = import ./profiles/bookmark-types.nix { inherit lib; };
+
+  sourceFile =
```

Comment:
I like the concept of this type, but agree it should be pulled out into the `lib` so other modules can take advantage of it.

Reply:
Do you think we should observe it in-practice for a while first? I promise, I won't forget about it.